src/java.base/share/classes/java/lang/foreign/VaList.java

```diff
@@ -87,6 +96,7 @@ sealed public interface VaList extends Addressable permits WinVaList, SysVVaList
      * {@linkplain MemorySession#isAlive() alive}.
      * @throws WrongThreadException if this method is called from a thread other than the thread owning
      * the {@linkplain #session() session} associated with this variable argument list.
+     * @throws NoSuchElementException if an out-of-bounds read is detected.
```

Comment:
I'd insert a link to the safety section, where the link text is "out-of-bounds".

Reply:
Good idea, I'll add links